src/gc.c

```diff
@@ -1579,17 +1580,19 @@ static jl_taggedvalue_t **sweep_page(jl_gc_pool_t *p, jl_gc_pagemeta_t *pg, jl_t
                     has_young = 1;
                 }
                 has_marked |= gc_marked(bits);
-                *ages |= msk;
+                age |= msk;
                 freedall = 0;
             }
             v = (jl_taggedvalue_t*)((char*)v + osize);
             msk <<= 1;
             if (!msk) {
                 msk = 1;
+                *ages = age;
                 ages++;
+                age = *ages;
             }
         }
-
+        *ages = age;
```

Comment:
This might be one past the end?
```suggestion
        if (msk != 1)
            *ages = age;
```

Reply:
I guess if we hit the `if(!msk)` branch on the last iteration this is possibly not legal, but that would mean `age = *ages;` a couple lines above is also possibly illegal

Reply:
Move that to the top of the for loop then?